public/js/clients/firefox.js

```diff
@@ -9,6 +9,58 @@ let currentClient = null;
 let currentThreadClient = null;
 let currentTabTarget = null;
 
+// API implementation
+
+let APIClient = {
+  resume() {
+    return new Promise(resolve => {
+      currentThreadClient.resume(resolve);
+    });
+  },
+
+  stepIn() {
+    return new Promise(resolve => {
+      currentThreadClient.stepIn(resolve);
+    });
+  },
+
+  stepOver() {
+    return new Promise(resolve => {
+      currentThreadClient.stepOver(resolve);
+    });
+  },
+
+  stepOut() {
+    return new Promise(resolve => {
+      currentThreadClient.stepOut(resolve);
+    });
+  },
+
+  getSources() {
+    return currentThreadClient.getSources();
+  },
+
+  sourceContents(sourceId) {
+    const sourceClient = currentThreadClient.source({ actor: sourceId });
+    return sourceClient.source();
+  },
+
+  setBreakpoint(location, condition) {
+    const sourceClient = currentThreadClient.source({ actor: location.sourceId });
+    return sourceClient.setBreakpoint({
+      line: location.line,
+      column: location.column,
+      condition: condition
+    });
+  }
+};
+
+function getAPIClient() {
+  return APIClient;
+}
```

Comment:
in terms of naming/location. I would like to put the api clients in their own file at some point (maybe now). 

It might be helpful if the api, protocol, and bootstrap file in client/chrome were moved to vendor and the build-bootstrap script were moved to bin. I think they belong their anyway as they are 3rd party code. 

If we did that then, I could see a clients/chrome clients/firefox directory having api modules.